src/DynamoDbCollection.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace BaoPham\DynamoDb;
+
+use Illuminate\Database\Eloquent\Collection;
+
+class DynamoDbCollection extends Collection
+{
+    protected $lastEvaluatedKey = null;
+    protected $conditionIndexKeys = null;
+    protected $model = null;
+
+    public function __construct(array $models = [], $lastEvaluatedKey = null, $conditionIndexKeys = null)
+    {
+        parent::__construct($models);
+        $this->lastEvaluatedKey = $lastEvaluatedKey;
+        $this->conditionIndexKeys = $conditionIndexKeys;
+
+        if (!$this->isEmpty()) {
+            $class = get_class($this->first());
+            $this->model = new $class;
+        }
+    }
+
+    public function getLastEvaluatedKey()
+    {
+        if (!empty($this->lastEvaluatedKey)) {
+            return $this->lastEvaluatedKey;
+        }
+
+        $after = $this->last();
+
+        if (empty($after)) {
+            return null;
+        }
+
+        $afterKey = $after->getKeys();
+
+        if ($this->conditionIndexKeys) {
+            $columns = array_values($conditionIndexKeys['keysInfo']);
+            foreach ($columns as $column) {
+                $afterKey[$column] = $after->getAttribute($column);
+            }
+        }
+
+        return $this->model->unmarshalValue($this->getDynamoDbKey($afterKey));
+    }
+
+    /**
+     * Return key for DynamoDb query.
+     *
+     * @param array|null $modelKeys
+     * @return array
+     *
+     * e.g.
+     * [
+     *   'id' => ['S' => 'foo'],
+     * ]
+     *
+     * or
+     *
+     * [
+     *   'id' => ['S' => 'foo'],
+     *   'id2' => ['S' => 'bar'],
+     * ]
+     */
+    protected function getDynamoDbKey($modelKeys = null)
```

Comment:
I like the idea of this class 👍 . But I'm unsure about this method being here. It seems out of place. It's also a copy and paste from the query builder so we have duplicates here.

Reply:
Correct - this class was an after thought, threw it in quickly to get the tests written and passed.  Not sure how to address this at the moment.